packages/nvidia_nat_vanna/src/nat/plugins/vanna/db_utils.py

```
@@ -0,0 +1,298 @@
+        # Store engine in Vanna instance - lifecycle matches singleton
+        vn.db_engine = engine
+        logger.info(f"Created and stored database engine in Vanna instance for {database_type}")
+
+    # Define async run_sql function for Vanna
+    async def run_sql(sql_query: str) -> pd.DataFrame:
+        """Execute SQL asynchronously and return DataFrame."""
+        try:
+            query_result = await async_execute_query(engine, sql_query)
+            return query_result.to_dataframe()
+        except Exception as e:
+            logger.error(f"Error executing SQL: {e}")
```

Ruff (0.14.3)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)